rust/lon/src/git.rs

```diff
@@ -129,6 +130,34 @@ fn find_newest_revision_for_ref(url: &str, reference: &str) -> Result<Revision>
     Ok(Revision(references.remove(0).revision))
 }
 
+/// Find the default branch for a git repository
+pub fn find_default_branch(url: &str) -> Result<String> {
+    let output = Command::new("git")
+        .arg("ls-remote")
+        .args(["--symref", url, "HEAD"])
+        .output()
+        .context("Failed to execute git ls-remote. Most likely it's not on PATH")?;
+
+    if !output.status.success() {
+        bail!(
+            "Failed to find the default branch for {}\n{}",
+            url,
+            String::from_utf8_lossy(&output.stderr)
+        )
+    }
+
+    let re = Regex::new(r"ref:.*refs/heads/(?<branch>.*)\tHEAD")?;
```

Comment:
I think we could achieve the same with some more naive whitespace splitting and prefix trimming, right?

Would also be nice to add a simple expect based test for the git output parsing (see https://github.com/nikstur/lon/blob/main/rust/lon/src/git.rs#L504)